core/trino-main/src/main/java/io/trino/operator/join/JoinUtils.java

```diff
@@ -78,6 +85,50 @@ public static boolean isBuildSideReplicated(PlanNode node)
                 .matches();
     }
 
+    public static Map<DynamicFilterId, Symbol> getJoinDynamicFilters(JoinNode joinNode, boolean taskRetriesEnabled)
+    {
+        Map<DynamicFilterId, Symbol> dynamicFilters = joinNode.getDynamicFilters();
+        if (!taskRetriesEnabled) {
```

Comment:
Why is this flag necessary? Why not to always try to recurse? (same for `getSemiJoinDynamicFilterId`)